source/store.go

```diff
@@ -16,28 +16,62 @@ limitations under the License.
 
 package source
 
-import "fmt"
+import "errors"
 
-var store = map[string]Source{}
+func init() {
+	// Register all selectable sources under an identifier.
+	RegisterFunc("service", NewServiceSource)
```

Comment:
another question is though how to avoid duplicating client in multiple places, but I don't see this as critical.